### PR TITLE
Fix #124 - Add support for zio environment

### LINKED ISSUE #124: Add support for ZIO environment
Many methods only support `IO`, not `ZIO`.

### PR DESCRIPTION
**Pull request by FrancisToth**
Modified Supervisor in order to accept an R as environment and changed Supervisor::retryOrElse to support that.

Also ignored files related to Metals configuration.

This resolves #124

### CODE REVIEW

File: src/main/scala/zio/actors/ActorSystem.scala [hunk 1]
```diff
@@ -138,7 +138,7 @@ final class ActorSystem private[actors] (
    */
   def make[S, E <: Throwable, F[+_]](
     actorName: String,
-    sup: Supervisor[E],
+    sup: Supervisor[Any, E],
```
Comment (jdegoes) [inline src/main/scala/zio/actors/ActorSystem.scala]: Should be `R`, right?

File: src/main/scala/zio/actors/ActorSystem.scala [hunk 2]
```diff
@@ -70,7 +70,7 @@ final class Context private[actors] (
    */
   def make[S, E1 <: Throwable, F1[+_]](
     actorName: String,
-    sup: Supervisor[E1],
+    sup: Supervisor[Any, E1],
```
Comment (jdegoes) [inline src/main/scala/zio/actors/ActorSystem.scala]: Should be `R`, right?

File: src/main/scala/zio/actors/Actor.scala [hunk 3]
```diff
@@ -32,7 +32,7 @@ object Actor {
   private val DefaultActorMailboxSize = 10000
 
   private[actors] final def stateful[S, E <: Throwable, F[+_]](
-    supervisor: Supervisor[E],
+    supervisor: Supervisor[Any, E],
```
Comment (jdegoes) [inline src/main/scala/zio/actors/Actor.scala]: R should be allowed here. Also for Actor.Stateful.

**Comment by FrancisToth** [comment]
@jdegoes  I was not sure about how big the scope for this change should be. I have added `-R` to all the places you have requested. This resulted in some significant changes in the API, but maybe we can address this later on by providing a more specialized one.

**Comment by jdegoes** [comment]
@FrancisToth Nice work! 💪

**Pull request merged**